docs/source/notes/early_stopping.rst

```diff
@@ -0,0 +1,115 @@
+|:octagonal_sign:| Early Stopping
+=================================
+
+Early stopping and threshold stopping halt training based on set criteria. In Composer, they are callbacks can be passed to the Trainer.
+
+
+Early Stopping
+--------------
+
+The :class:`.EarlyStopper` callback stops training if a provided metric does not improve over a certain ``patience`` window of time.
+
+.. testcode::
+
+    import torch
+    from composer import Trainer
+    from composer.callbacks.early_stopper import EarlyStopper
+
+    early_stopper = EarlyStopper(
+        monitor='Accuracy',
+        dataloader_label='train',
+        patience='50ba',
+        comp=torch.greater,
+        min_delta=0.01,
+    )
+
+    trainer = Trainer(
+        model=model,
+        train_dataloader=train_dataloader,
+        eval_dataloader=eval_dataloader,
+        optimizers=optimizer,
+        callbacks=[early_stopper],
+        max_duration="1ep",
+    )
+
+In the above example, the ``'train'`` label means the callback is tracking the ``Accuracy`` metric for the train_dataloader. The default for the evaluation dataloader is ``eval``.
+
+We also set ``patience='50ba'`` and ``min_delta=0.01`` which means that every 50 batches, if the Accuracy does not exceed the best recorded Accuracy by ``0.01``, training is stopped. The `comp` argument indiciates that 'better' here means higher accuracy.
```

Comment:
[nit] might be helpful to mention here that `patience` can be specified in different time units.  Or just point to the time abstraction for details.